app/main.py

```diff
@@ -2,0 +1,48 @@
+class Animal:
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def die(self) -> None:
+        Animal.alive.remove(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+    
+    @classmethod
+    def print_alive(cls) -> None:
+        return [
+            {
+                "Name": animal.name,
+                "Health": animal.health,
+                "Hidden": animal.hidden
+            } for animal in cls.alive
+        ]
+
+
+class Herbivore(Animal):
+    def hide(self) -> bool:
+        self.hidden = not self.hidden
+        return self.hiden   
+
+
+class Carnivore(Animal):
+    def bite(self, victim: Herbivore) -> bool:
+        if isinstance(victim, Carnivore):
+            print('Carnivore cannot bite by another carnivore!')
+        elif not victim.hiden:
+            victim.health -= 50
+            if victim.health <= 0:
+                victim.health = 0
+                victim.die()
+        elif victim.hidden:
+            victim.health = 100
```

Comment:
The logic here seems incorrect. If the victim is hidden, their health should not be reset to 100. Consider revising this logic to reflect the intended behavior when a hidden victim is bitten.